lib/model/autocomplete.dart

```diff
@@ -163,6 +183,48 @@ final RegExp _emojiIntentRegex = (() {
     + r')$');
 })();
 
+final RegExp _channelLinkIntentRegex = () {
+  // What's likely to come just before #channel syntax: the start of the string,
+  // whitespace, or punctuation. Letters are unlikely; in that case a GitHub-
+  // style "zulip/zulip-flutter#124" link might be intended (as on CZO where
+  // there's a custom linkifier for that).
+  //
+  // By punctuation, we mean *some* punctuation, like "(". We make "#" and "@"
+  // exceptions, to support typing "##channel" for the channel query "#channel",
+  // and typing "@#user" for the mention query "#user", because in 2025-11
+  // channel and user name words can start with "#". (They can also contain "#"
```

Comment:
nit:
```suggestion
  // and typing "@#user" for the mention query "#user", because as of 2025-11
  // channel and user name words can start with "#". (They can also contain "#"
```

Otherwise it sounds like you're suggesting it was different before, or will be different later. With "as of", it means it's this way now and you're just not saying anything about other times. The code is effectively presuming it's that way at all times.